Sources/Plasma/NucleusLib/pnNetProtocol/Private/pnNpCommon.cpp

```diff
@@ -297,605 +294,289 @@ void NetGameRank::CopyFrom(const NetGameRank & fromRank) {
-        WRITE(kUuid_4,          IWriteValue,    uuid_4          );
-        WRITE(kString64_1,      IWriteString,   string64_1 ? string64_1 : L""       );
-        WRITE(kString64_2,      IWriteString,   string64_2 ? string64_2 : L""       );
-        WRITE(kString64_3,      IWriteString,   string64_3 ? string64_3 : L""       );
-        WRITE(kString64_4,      IWriteString,   string64_4 ? string64_4 : L""       );
-        WRITE(kString64_5,      IWriteString,   string64_5 ? string64_5 : L""       );
-        WRITE(kString64_6,      IWriteString,   string64_6 ? string64_6 : L""       );
-        WRITE(kIString64_1,     IWriteString,   istring64_1 ? istring64_1 : L""     );
-        WRITE(kIString64_2,     IWriteString,   istring64_2 ? istring64_2 : L""     );
-        WRITE(kText_1,          IWriteString,   text_1 ? text_1 : L""               );
-        WRITE(kText_2,          IWriteString,   text_2 ? text_2 : L""               );
-        WRITEARR(kBlob_1,       IWriteArray,    blob_1, blob_1Length);
-        WRITEARR(kBlob_2,       IWriteArray,    blob_2, blob_2Length);
-    #undef WRITEARR
-    #undef WRITE
-
-    if (flags & ~kAllValidFields)
-        FATAL("Invalid field flag(s) encountered");
-
-    if (rwOpts & kRwUpdateDirty)
-        dirtyFlags = 0;
-    // else, preserve existing dirtyFlags value
-
-    return buffer->Count() - pos;
+bool NetVaultNode::Blob::operator==(const Blob& rhs) const
+{
+    if (size == rhs.size)
+        return memcmp(buffer, rhs.buffer, size) == 0;
+    return false;
 }
 
 //============================================================================
-bool NetVaultNode::Matches (const NetVaultNode * other) {
+void NetVaultNode::Clear()
+{
+    // Sneaky -- we're just going to set the fields to empty.
+    // If a field is neither used nor dirty, it doesn't matter what value it actually has.
+    fUsedFields = 0;
+    fDirtyFields = 0;
+    fRevision = kNilUuid;
+}
+
+//============================================================================
+void NetVaultNode::CopyFrom(const NetVaultNode* node)
+{
+    fUsedFields = node->fUsedFields;
+    fDirtyFields = node->fDirtyFields;
+    fRevision = node->fRevision;
+
+#define COPY(field) f##field = node->f##field
+    COPY(NodeId);
+    COPY(CreateTime);
+    COPY(ModifyTime);
+    COPY(CreateAgeName);
+    COPY(CreateAgeUuid);
+    COPY(CreatorAcct);
+    COPY(CreatorId);
+    COPY(NodeType);
+    COPY(Int32_1);
+    COPY(Int32_2);
+    COPY(Int32_3);
+    COPY(Int32_4);
+    COPY(UInt32_1);
+    COPY(UInt32_2);
+    COPY(UInt32_3);
+    COPY(UInt32_4);
+    COPY(Uuid_1);
+    COPY(Uuid_2);
+    COPY(Uuid_3);
+    COPY(Uuid_4);
+    COPY(String64_1);
+    COPY(String64_2);
+    COPY(String64_3);
+    COPY(String64_4);
+    COPY(String64_5);
+    COPY(String64_6);
+    COPY(IString64_1);
+    COPY(IString64_2);
+    COPY(Text_1);
+    COPY(Text_2);
+    COPY(Blob_1);
+    COPY(Blob_2);
+#undef COPY
+}
```

Comment:
It looks like the old version zeroed fields not set in the bitmask...  I feel like we should keep this behavior in order to avoid misbehaving apps which don't check the bitmask before using a value, or else check the bitmask in the various getter functions.